### ISSUE TITLE
Postgres connection problem

### ISSUE THREAD

**Issue by prg** [jberger/Minion-Notifier#3]
```perl
#!/usr/bin/env perl
use Mojolicious::Lite;
 
plugin Minion => { Pg => 'postgresql://<user>:<pw>@/test' };
 
app->minion->add_task(foo => sub {
    my $job = shift;
    $job->finish({ bar => 'baz' });
});
 
plugin 'Minion::Notifier';
 
get '/foo' => sub {
    my $c = shift;
    foreach my $i (0 .. 9) {
        $c->minion->enqueue(foo => [id => $i]);
    }
    $c->render(text => 'Jobs queued');
};
 
app->start;
```

- enqueue jobs with `myapp.pl get /foo`
- start worker with `myapp.pl minion worker`

Worker output:
```
[Fri Mar  3 14:05:37 2017] [debug] Performing job "2" with task "foo" in process 9351
message type 0x43 arrived from server while idle
message type 0x5a arrived from server while idle
DBD::Pg::db do warning:  at /home/<...>/5.24.0/Mojo/Pg/Database.pm line 59.
```

Postgres log says `DEBUG:  unexpected EOF on client connection`.

A couple of observations:
- doesn't occur when replacing `$job->finish` with `return`
- doesn't occur when removing `plugin 'Minion::Notifier';` (`$job->finish` works fine in that case)
- worker processes can only be terminated with SIGKILL

In my real application things are even worse: Minion worker just exits after a couple of jobs, database connections randomly stop working.

Module versions:
```
Mojolicious
-------------------------------------------------------------------------
        Installed: 7.27
        CPAN:      7.27  up to date
Minion
-------------------------------------------------------------------------
        Installed: 6.02
        CPAN:      6.02  up to date
Minion::Notifier
-------------------------------------------------------------------------
        Installed: 0.05
        CPAN:      0.05  up to date
Mojo::Pg
-------------------------------------------------------------------------
        Installed: 3.02
        CPAN:      3.02  up to date
DBD::Pg
-------------------------------------------------------------------------
        Installed: 3.5.3
        CPAN:      3.5.3  up to date
```
Postgres:
```
                                                     version
-----------------------------------------------------------------------------------------------------------------
 PostgreSQL 9.5.5 on x86_64-pc-linux-gnu, compiled by gcc (Ubuntu 5.4.0-6ubuntu1~16.04.2) 5.4.0 20160609, 64-bit
```
Running perlbrew `perl-5.24.0`, if you think it might be relevant I can provide `perl -V` output.

**Comment by prg** [comment]
Quick update:

I did a quick & dirty reimplementation of the `Pg` transport, and it seems to have the same problem:
```
Worker error: DBD::Pg::db do failed: SSL SYSCALL error: EOF detected at /home/<...>/5.24.0/Mojo/Pg/Database.pm line 67.
```

Maybe something else is to blame...

**Comment by jberger** [comment]
This is (or should be) fixed in 46259fd9aeb7dededd4f29743489beee5bf7f640 .

Thanks for the report!
